beacon-chain/node/node.go

```diff
@@ -521,6 +523,16 @@ func (b *BeaconNode) registerP2P(cliCtx *cli.Context) error {
 		return err
 	}
 
+	var activeVals uint64
+	st := b.finalizedStateAtStartUp
+	if st == nil || st.IsNil() {
+		log.Warn("Starting p2p with active validator count = 0. This should only happen with networks in a pre-genesis state")
```

Comment:
want to highlight this @nisdas - using val count of zero when finalized state is not present. This also means genesis is  not in the DB. This happens during e2e tests because we start from deposits.

Reply:
btw I'm open to making this a `log.Error` instead of Warn. Since testnets going forward are likely going to all use pre-generated genesis states, I expect we're close to being able to deprecate this sort of pre-phase0 corner case. So it seems like a safe assumption that if a user sees this in their logs outside of e2e, it is an error.

Reply:
Does the finalized state cached in stategen always fallback to the genesis in the event there are no other states ?


The one difference that happens now is that, previously we would only register topics(and their scoring params) once genesis happened. So we would always have a validator count that we could use. Right now, if it is pre-genesis it seems like we are stuck at 0 for the whole run.

Reply:
Very good catch. As we discussed in discord I'll plan to address this by passing in a new type (something like `CurrentValidatorCounter`) instead of the uint value. If the finalized state in `BeaconNode` is non-nil, the new type can use it, otherwise fallback to the DB when the count is eventually requested by p2p.